### PR TITLE
roachtest: include branch in created issues

### PR DESCRIPTION
**Pull request by tbg**
This addresses part of
https://github.com/cockroachdb/cockroach/issues/43920.

In short, it's tedious to track failed roachtests on release branches
when the corresponding issues mingle freely with those from other
branches. Putting the release branch in the title is an easy step
in the right direction.

Release note: None

### CODE REVIEW

**Comment by tbg** [comment]
> Want the bonus points?

@dt's call if you ask me -- that will delay this PR quite a bit since it requires upfronting a larger refactor of `issues` and the roachtest issue template. I do like the idea.

**Comment by dt** [comment]
I love the bonus points suggestion but see no reason it should need to be included in or hold up merging the improvement here -- as-is, this PR is already an improvement over the current situation and adding linking can be done separately.

**Comment by tbg** [comment]
Ok, as far as I'm concerned I'm waiting for an LGTM here.

I will explore the cross-linking separately. However, my hunch is that it might be better to "just" link to a roachdash page that groups all roachtests properly because that is straightforward, and trying to cross-correlate issues manually is hard because all we have are the titles.

File: pkg/cmd/roachtest/test_runner.go [hunk 1]
```diff
@@ -619,10 +619,17 @@ func (r *testRunner) runTest(
...
 				artifacts := fmt.Sprintf("/%s", t.Name())
+				title := fmt.Sprintf("roachtest: %s failed", t.Name())
+				// Don't disambiguate by branch if the branch is master, to avoid
+				// creating duplicates with issues created prior to when this code
+				// was introduced.
+				if branch != "master" {
+					title = fmt.Sprintf("roachtest: %s failed on %s", t.Name(), branch)
+				}
 
 				req := issues.PostRequest{
```
Comment (dt) [inline pkg/cmd/roachtest/test_runner.go]: i'm unfamilar with this code -- where do we attempt to find and update an existing issue vs post a new one?

**Comment by andreimatei** [comment]
Without the crosslinking I'm not very convinced that this is an improvement. I'll let you continue without me.

File: pkg/cmd/roachtest/test_runner.go (hunk 1 above)
Comment (tbg) [reply pkg/cmd/roachtest/test_runner.go]: it's in `issues` and basically searches github for the title of the desired issue. See my comment in the main thread on how I'm planning to improve it.

**Review by dt — APPROVED** [review]

**Comment by tbg** [comment]
I think I know what to do here.

TLDR:

We achieve the per-branch distinction simply by having `roachtest` pass extra labels `<branchname>` and `release-blocker` (names tbd) when branch is not master. We need to teach issues to create labels as needed.

In issues, search GH for: `is:issue is:open in:title label:C-test-failure <testname>`

Note that this ignores the extra labels supplied by the caller, such as `O-roachtest`, intentionally.

Manually filter for exact word-match for the test name (GH is too lenient) and additional labels, name the first result (should be at most one) the target issue to post to (create if none)

Use the remaining matches to populate the list Andrei wants (and which is now handy and thus definitely something we want).

All release branch roachtests could then be found searching for `label:O-roachtest,release-blocker`. To restrict to a certain release, add `label:release-19.2`.

Thoughts?

**Comment by tbg** [comment]
PS @dt I'm holding off on merging because your question about duplicates made me take another look and the search is too fuzzy, I think the master issues could end up getting posted into the release branch issues, at which point it's better to hold off and execute on the plan above instead.

**Comment by tbg** [comment]
#44155
